specs/ExtendedNewWindowRequested.md

```diff
@@ -0,0 +1,78 @@
+# Background
+WebView2 raises the `NewWindowRequested` event when a new window is opened. The WebView2 team has been asked to include the name of the new window 
+being created as a parameter of `NewWindowRequestedEventArgs`. This window name is the name given to the window when it is opened 
+using `window.open(url, windowName)`.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending the `NewWindowRequestedEventArgs` to provide access to the `Name` property. 
+The `Name` property will return the name of the new window being created. 
+
+# Examples
+## C++: Get name of window
+
+``` cpp
+wil::com_ptr<ICoreWebView2> m_webviewEventSource;
+EventRegistrationToken m_newWindowRequestedToken = {};
+
+m_webviewEventSource->add_NewWindowRequested(
+    Callback<ICoreWebView2NewWindowRequestedEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args)
+          -> HRESULT {
+              wil::com_ptr<ICoreWebView2NewWindowRequestedEventArgs2> args2;
+              
+              if(SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args2)))) {
+                  wil::unique_cotaskmem_string name;
+                  CHECK_FAILURE(args2->get_Name(&name));
+              }
```

Comment:
The customer wants the name so they can have additional context about who/why the new window is being requested correct? Maybe we can add some code that does something like that?

```suggestion
                   if (wcscmp(name, L"openInNewBrowser") == 0)
                   {
                       ShellExecute(...); // I don't remember actually the params to ShellExecute
                       args2->put_Handled(true);
                   }
                   else
                   {
                       // normal new window handling...?
                   }
              }
```